src/javascript/app_2/App/Components/Elements/Calendar/calendar.jsx

```diff
@@ -0,0 +1,213 @@
+    }
+
+    // sets Calendar active view
+    calendarViews = {
+        date  : () => { this.setState({ calendar_view: 'date'   }); },
+        month : () => { this.setState({ calendar_view: 'month'  }); },
+        year  : () => { this.setState({ calendar_view: 'year'   }); },
+        decade: () => { this.setState({ calendar_view: 'decade' }); },
+    }
+
+    componentWillMount() {
+        this.setState({ calendar_view: 'date' });
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const date = moment(this.state.calendar_date);
+
+        if (date.isBefore(moment(nextProps.minDate))) {
+            this.setState({
+                date: nextProps.minDate,
+            });
+        } else if (date.isAfter(moment(nextProps.maxDate))) {
+            this.setState({
+                date: nextProps.maxDate,
+            });
+        }
+    }
+
+    navigateTo = (value, unit, is_add) => {
+        const { dateFormat, maxDate, minDate } = this.props;
+
+        let new_date = moment(this.state.calendar_date, dateFormat)[is_add ? 'add' : 'subtract'](value, unit).format(dateFormat);
+
+        if (unit === 'months' && this.isPeriodDisabled(new_date, 'month')) return;
+
+        if (unit === 'years'  && this.isPeriodDisabled(new_date, 'years')) {
+            new_date = is_add ? maxDate : minDate;
+        }
+
+        this.setState({ calendar_date: moment(new_date, dateFormat).format(dateFormat) }); // formatted date
+    }
+
+    updateSelectedDate = (e) => {
+        const { dateFormat, maxDate, minDate, onSelect } = this.props;
+
+        const date      = moment(e.target.dataset.date);
+        const min_date  = moment(minDate).format(dateFormat);
+        const max_date  = moment(maxDate).format(dateFormat);
+        const is_before = date.isBefore(min_date);
+        const is_after  = date.isAfter(max_date);
+
+        if (is_before || is_after) return;
+
+        const formatted_date = date.format(dateFormat);
+        this.setState({
+            calendar_date: formatted_date,
+            selected_date: formatted_date,
+        });
+
+        if (onSelect) {
+            onSelect(formatted_date);
+        }
+    }
+
+    updateSelected = (e, type) => {
+        const view_map = {
+            month : 'date',
+            year  : 'month',
+            decade: 'year',
+        };
+        const date = moment(this.state.calendar_date, this.props.dateFormat)[type === 'decade' ? 'year' : type](e.target.dataset[type].split('-')[0]).format(this.props.dateFormat);
+
+        if (this.isPeriodDisabled(date, type)) return;
+
+        this.setState({
+            calendar_date: date,
+            calendar_view: view_map[type],
+        });
+    }
+
+    resetCalendar = () => {
+        const { dateFormat, startDate } = this.props;
+
+        const default_date = (startDate ? moment(startDate) : moment()).utc().format(dateFormat);
+        this.setState({
+            calendar_date: default_date,
+            selected_date: '',
```

Comment:
@kellybinary 
it's better to set `calendar_view` to its default value when the calendar is reset.